mesonbuild/compilers/compilers.py

```diff
@@ -998,12 +998,12 @@ def get_pic_args(self) -> T.List[str]:
         m = 'Language {} does not support position-independent code'
         raise EnvironmentException(m.format(self.get_display_language()))
 
-    def get_pie_args(self) -> T.List[str]:
+    def get_pie_args(self, env: 'Environment') -> T.List[str]:
```

Comment:
It is only imported if `TYPE_CHECKING`, so it does. See for example:

```python
    def get_define(self, dname: str, prefix: str, env: 'Environment',
                   extra_args: T.Union[T.List[str], T.Callable[[CompileCheckMode], T.List[str]]],
                   dependencies: T.List['Dependency'],
                   disable_cache: bool = False) -> T.Tuple[str, bool]:
        raise EnvironmentException('%s does not support get_define ' % self.get_id())
```

Reply:
It is not needed but we haven't done a full treewide rewrite to drop two unneeded bytes on most lines.

Due to the use of `from __future__ import annotations` all type annotations are implicitly stringized, regardless of whether they are quoted in the source code. It is much, much, much easier than always trying to remember whether or not to quote, and it means that the bytecode compiler produces faster bytecode even for stuff like `T.List['Dependency']` since the List object isn't evaluated to produce a code object.

Yes, even though List is available at runtime because the typing module is imported in order to check TYPE_CHECKING, we don't want that to be executed at runtime either.

Inside of `T.cast()` it's still necessary to string-quote type annotations as function argument values.

Reply:
Ok, I stand corrected!